model/resolver/resolver.go

```diff
@@ -345,6 +345,43 @@ func (r *Resolver) ResolveLinks() validation.ErrorList {
 		}
 	}
 
+	// Record in each job what roles consume it
+	for _, consumerInstanceGroup := range m.InstanceGroups {
```

Comment:
What do you think about pulling the logic from the inner-most loop into a function?